XPlat/Gigya/Android/source/GigyaSDK.Android/Additions.cs

```diff
@@ -0,0 +1,242 @@
+			SendRequest(method, @params, useHTTPS, new GSRespnseActionListener(tcs), null);
+			return tcs.Task;
+		}
+	}
+
+	internal class GSPermissionResultActionHandler : Java.Lang.Object, IGSPermissionResultHandler
+	{
+		private readonly TaskCompletionSource<PermissionsResult> tcs;
+
+		public GSPermissionResultActionHandler(TaskCompletionSource<PermissionsResult> tcs)
+		{
+			this.tcs = tcs;
+		}
+
+		public void OnResult(bool granted, Java.Lang.Exception exception, IList<string> declinedPermissions)
+		{
+			if (exception != null)
+			{
+				tcs.SetException(exception);
+			}
+			else
+			{
+				var result = new PermissionsResult(granted, declinedPermissions.ToArray());
+				tcs.SetResult(result);
+			}
+		}
+	}
+
+	public struct PermissionsResult
+	{
+		public PermissionsResult(bool granted, string[] declinedPermissions)
+		{
+			Granted = granted;
+			DeclinedPermissions = declinedPermissions;
+		}
+
+		public bool Granted { get; private set; }
+		public string[] DeclinedPermissions { get; private set; }
+	}
+}
+
+namespace GigyaSDK.Socialize.Android.Events
+{
+	internal class GSAndroidPermissionActionListener : Java.Lang.Object, IGSAndroidPermissionListener
+	{
+		private readonly TaskCompletionSource<AndroidPermissionsResult> tcs;
+
+		public GSAndroidPermissionActionListener(TaskCompletionSource<AndroidPermissionsResult> tcs)
+		{
+			this.tcs = tcs;
+		}
+
+		internal int RequestCode { get; set; }
+
+		public void OnAndroidPermissionsResult(string[] permissions, Permission[] grantResults)
+		{
+			var result = new AndroidPermissionsResult(RequestCode, permissions, grantResults);
+			tcs.SetResult(result);
+		}
+	}
+
+	public struct AndroidPermissionsResult
+	{
+		public AndroidPermissionsResult(int requestCode, string[] permissions, Permission[] grantResults)
+		{
+			RequestCode = requestCode;
+			Permissions = permissions;
+			GrantResults = grantResults;
+		}
+
+		public int RequestCode { get; internal set; }
+		public string[] Permissions { get; private set; }
+		public Permission[] GrantResults { get; private set; }
+	}
+}
+
+namespace GigyaSDK.Socialize.Android.Login.UI
+{
+	[Activity(
+		Theme = "@android:style/Theme.Translucent.NoTitleBar",
+		ConfigurationChanges = ConfigChanges.Keyboard | ConfigChanges.KeyboardHidden | ConfigChanges.ScreenLayout | ConfigChanges.ScreenSize | ConfigChanges.Orientation)]
+	partial class HostActivity
+	{
+	}
+}
+
+namespace GigyaSDK.Socialize.Android.Login.Providers
+{
+	[Activity(
+		Theme = "@android:style/Theme.Translucent.NoTitleBar",
+		LaunchMode = LaunchMode.SingleTask,
+		AllowTaskReparenting = true,
+		ConfigurationChanges = ConfigChanges.Keyboard | ConfigChanges.KeyboardHidden | ConfigChanges.ScreenLayout | ConfigChanges.ScreenSize | ConfigChanges.Orientation)]
+	[IntentFilter(
+		new[] { Intent.ActionView },
+		Categories = new[] { Intent.CategoryDefault, Intent.CategoryBrowsable },
+		DataScheme = "@PACKAGE_NAME@", DataHost = "gsapi")]
+	partial class WebLoginActivity
```

Comment:
This partial class with attributes will automatically add the bits to the AndroidManifest.xml. 
Thus, this step can be skipped: http://developers.gigya.com/display/GD/Android#Android-AddtoAndroidManifest.xml